api/server/secrets.go

```diff
@@ -0,0 +1,125 @@
+package server
+
+import (
+	"encoding/json"
+	"net/http"
+	"strconv"
+
+	"github.com/libopenstorage/openstorage/secrets"
+)
+
+const (
+	secretKeyOkMsg   = "Secret Key set successfully."
+	secretLoginOkMsg = "Secrets Login Succeeded"
+	secretLoginCheck = "Secrets Login Check Succeeded"
+)
+
+//TODO: Add swagger yaml
+func (c *clusterApi) setClusterSecretKey(w http.ResponseWriter, r *http.Request) {
+
+	params := r.URL.Query()
```

Comment:
Do not use URL Query. Instead either get a JSON packet from the request or use mux.Vars() to get any vars in the URL, like ID.